huaweicloud/validators.go

```diff
@@ -95,6 +95,42 @@ func validateCIDR(v interface{}, k string) (ws []string, errors []error) {
 	return
 }
 
+func validateIPRange(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	ipAddresses := strings.Split(value, "-")
+	if len(ipAddresses) != 2 {
+		errors = append(errors, fmt.Errorf(
+			"%q must be a valid network IP address range, such as 0.0.0.0-255.255.255.0, but got %q", k, value))
+		return
+	}
+	netIPs := make([]net.IP, 0, 2)
+	for _, address := range ipAddresses {
+		ipnet := net.ParseIP(address)
+		if ipnet == nil || address != ipnet.String() {
+			errors = append(errors, fmt.Errorf("%q must contains valid network IP address, got %q", k, address))
+		} else {
+			netIPs = append(netIPs, ipnet)
+		}
+	}
+	if len(errors) == 0 {
+		if netIPs[0].Equal(netIPs[1]) {
+			errors = append(errors, fmt.Errorf("Two network IP address of %q cannot equal, got %q", k, value))
+		}
+		ipIndexArr := []int{12, 13, 14, 15}
+		for _, index := range ipIndexArr {
```

Comment:
s/ipIndexArr/len(netIPs[0])/

Reply:
Use a for loop to compare the four parts of IPv4 address in turn now.
It's easy to read than range 12 to 15 of net.IP.
Split string to arrays have less time consuming.